tests/vanilla/dependency.test.tsx

```diff
@@ -274,3 +274,43 @@ it('refreshes deps for each async read', async () => {
   resolve.splice(0).forEach((fn) => fn())
   expect(values).toEqual([0, 1])
 })
+
+it('handles errors in async atoms', async () => {
+  const errorAtom = atom(async () => {
+    throw new Error('Test error')
+  })
+
+  const store = createStore()
+
+  await expect(store.get(errorAtom)).rejects.toThrow('Test error')
+})
+
+it('handles complex dependency chains', async () => {
+  const baseAtom = atom(1)
+  const derived1 = atom((get) => get(baseAtom) * 2)
+  const derived2 = atom((get) => get(derived1) + 1)
+  const asyncDerived = atom(async (get) => {
+    const value = get(derived2)
+    await new Promise((resolve) => setTimeout(resolve, 10))
+    return value * 2
+  })
+
+  const store = createStore()
+  expect(await store.get(asyncDerived)).toBe(6)
+
+  store.set(baseAtom, 2)
+  expect(await store.get(asyncDerived)).toBe(10)
+})
+
+it('handles large number of atoms efficiently', () => {
+  const store = createStore()
+  const atoms = new Array(1000).fill(null).map((_, i) => atom(i))
+  const derivedAtom = atom((get) => atoms.reduce((sum, a) => sum + get(a), 0))
+
+  const start = performance.now()
```

Comment:
Hmm, I don't think we do performance tests in our unit tests.
Can you try something with `./benchmarks`? It's not a test that runs on CI, but you can try more complex cases and share the result in a discussion.

Reply:
I think the existing benchmark suite also needs some refinement. I did some resource tracking a while back and memory consumption was not lining up with what I thought the tests are testing for

Reply:
> I think the existing benchmark suite also needs some refinement.

I actually agree. Not just refinement, you can create a brand new benchmark suite. We can even delete existing one, if yours covers it. That would be pretty helpful.

Reply:
@backbone87 Appreciated for sharing detections you noticed before. @dai-shi I prefer to leave this topic aside and it should independently handle by another clear and goal-focused pull request. These changes have been removed at this moment. 🙇